src/serialize.h

```diff
@@ -513,6 +513,28 @@ struct VarIntFormatter
     }
 };
 
+template<int Bytes>
+struct CustomUintFormatter
+{
+    static_assert(Bytes > 0 && Bytes <= 8, "CustomUintFormatter Bytes out of range");
+    static constexpr uint64_t MASK = 0xffffffffffffffff >> (8 * (8 - Bytes));
+
+    template <typename Stream, typename I> void Ser(Stream& s, I v)
+    {
+        if (v < 0 || (v & ~I(MASK)) != 0) throw std::ios_base::failure("CustomUintFormatter value out of range");
```

Comment:
In commit "Add CustomUintFormatter" (96b18829d750baea8577cf409d85c0cd5186b67c)

Could replace `(v & ~I(MASK)) != 0` with `v <= MASK` to be more consistent with other assert and avoid making an assumption about representation of `v`.

Reply:
Done, also renamed MASK to MAX.